src/sg/flybot/pullable/pattern.clj

```diff
@@ -70,5 +73,79 @@
        (pattern-error! "unable to understand" pattern)))))
 
 (comment
-  (->query #(concat % ['ok]) '{(:a :with [{:b 2 :c 3}]) {:b ?}})
+  (->query #(concat % ['ok]) '{(:a :with [{:b 2 :c 3}]) {:b ?}}) 
   )
+
+(defn valid-symbol?
+  "Returns true if `p` is a symbol starting with ?"
+  [p]
+  (and (symbol? p) (re-matches #"\?.*" (name p))))
+
+
+(def sch-pattern-proc
+  "Malli schema for the different post-processors in the keys.
+   Expects individual post-processors with format [proc arg]
+   i.e. [:not-found 0]"
+  [:multi {:dispatch first}
+   [:when [:tuple [:= :when] fn?]]
+   [:not-found [:tuple [:= :not-found] :any]]
+   [:with [:tuple [:= :with] vector?]]
```

Comment:
`vector?` can be more specific.

Reply:
We don't know the number of args neither their types, it could be anything no?

We don't know the function the args are passed to (could `inc`, `str` etc)